controller_manager/controller_manager/spawner.py

```diff
@@ -248,22 +198,16 @@ def main(args=None):
             controller_manager_name = f"/{controller_manager_name}"
 
     try:
-        if not wait_for_controller_manager(
-            node, controller_manager_name, controller_manager_timeout
-        ):
-            node.get_logger().error(
-                bcolors.FAIL + "Controller manager not available" + bcolors.ENDC
-            )
-            return 1
-
         for controller_name in controller_names:
             fallback_controllers = args.fallback_controllers
             controller_type = args.controller_type
             prefixed_controller_name = controller_name
             if controller_namespace:
                 prefixed_controller_name = controller_namespace + "/" + controller_name
 
-            if is_controller_loaded(node, controller_manager_name, prefixed_controller_name):
+            if is_controller_loaded(
+                node, controller_manager_name, prefixed_controller_name, controller_manager_timeout
```

Comment:
@fmauch do you know if this timeout needs to be added to the other controller manager service callback as well?

Reply:
"Needs to" is debatable. Basically, if we assume that either none or all controller manager services are available because we do not know whether the controller manager is started, it is sufficient to wait for the first called service only (which is, in fact, the `list_controllers` service, which is called first in `is_controller_loaded`).

With the proposed implementation we would timeout and exit if a timeout is given or wait until it is available if none is given. For all following service calls we will wait indefinitely, which should be fair under the assumption that all services should be available once the `list_controllers` service was available.

In my opinion, it is fair to leave it like that, though it might be a bit implicit.

BTW, that line seems to be the only merge conflict at the moment.

Reply:
Makes total sense. Let's leave it like you said!